### PR TITLE
Adding JWT Authentication to CodeIgniter Rest Server

### PR DESCRIPTION
**Pull request by aiamk**
it will add value to rest server as a JWT branch

Thanks

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# CodeIgniter Rest Server
+# CodeIgniter Rest Server with JWT Authentication
```
Comment (chriskacerguis) [inline README.md]: It's still just the CI Rest Server :)

File: application/config/rest.php [hunk 2]
```diff
@@ -187,6 +199,12 @@
 // ---Uncomment list line for the wildard unit test
 // $config['auth_override_class_method']['wildcard_test_cases']['*'] = 'basic';
 
+// JWT Exmaple
```
Comment (chriskacerguis) [inline application/config/rest.php]: So using these config is not a good idea, as they are used for something different.  We should not use them (maybe new config vars)

File: application/libraries/REST_Controller.php [hunk 3]
```diff
@@ -1,7 +1,12 @@
 <?php
-
 defined('BASEPATH') OR exit('No direct script access allowed');
 
+require APPPATH . '/libraries/jwt/JWT.php';
+require APPPATH . '/libraries/jwt/BeforeValidException.php';
```
Comment (chriskacerguis) [inline application/libraries/REST_Controller.php]: Shouldn't these all be included in the JWT.php file?

File: README.md (hunk 1 above)
Comment (aiamk) [inline README.md]: Hi  chriskacerguis,
its my first time to form a repository ,i will fix it.
thank you

File: application/config/rest.php (hunk 2 above)
Comment (aiamk) [inline application/config/rest.php]: i will think about it

File: application/libraries/REST_Controller.php (hunk 3 above)
Comment (aiamk) [inline application/libraries/REST_Controller.php]: you are right, i am afraid to use composer, some developers prefer to include the library.

**Comment by aiamk** [comment]
I thought the Example will make it more easier to understand the JWT Auth flow, and will give more flexibility to integrate it with any authentication library.

I will check the both scenario and revamp the code again.

Thank you for your valuable time.

**Pull request closed without merging**